### PR TITLE
DOCSP-35056 Dart upgrade dependencies

### PR DESCRIPTION
**Pull request by lindseymoore**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-35056

The following tests are not working:

- test/sync_multiple_processes_test.dart
- test/authenticate_users_test.dart
- test/manage_sync_session_test.dart
- test/open_flexible_sync_realm_test.dart

### CODE REVIEW

File: examples/dart/test_sdk_example/pubspec.lock
Comment (lindseymoore) [inline examples/dart/test_sdk_example/pubspec.lock:1]: Should I add back this deleted file?

Comment (dacharyc) [reply examples/dart/test_sdk_example/pubspec.lock:1]: Generally speaking, I would expect changes to this file if we do the load dependency command _inside_ this directory. But I know we had some challenges while we were on the call dealing with multiple pubspec.yaml files. It's probably not super important for this PR. We can always regenerate it in the future when we're working in this directory.

**Review by dacharyc — CHANGES_REQUESTED** [review]
It looks like there are a couple of things added here that we should remove - a few generated files, but more importantly, the `flutter` git submodule. After those changes, this should be good to go!

**Review by dacharyc — APPROVED** [review]
LGTM! Thanks for taking care of this, Lindsey!

**Pull request merged**